test/e2e/dnslookup_test.go

```diff
@@ -331,11 +355,20 @@ func writeDNSHammerFile(path string, records map[string][]string) error {
 	return csvWriter.WriteAll(data)
 }
 
-func generateTXTrecords(n int) map[string][]string {
+func generateCNAMErecords(n int, domain string) map[string][]string {
+	records := make(map[string][]string)
+	for i := 1; i <= n; i++ {
+		subDomain := strings.ToLower(randomTxt(rand.Intn(10-1) + 1)) // nolint: gosec
+		domain := fmt.Sprintf("%s-%d.test.", baseDomain, i)
+		records[subDomain+"."+domain] = []string{domain}
+	}
+	return records
+}
+
+func generateTXTrecords(n int, domain string) map[string][]string {
 	records := make(map[string][]string)
-	baseDomain := "host-resolver-e2e-test"
 	for i := 1; i <= n; i++ {
-		records[fmt.Sprintf("%s%d.test.", baseDomain, i)] = generateTXT(rand.Intn(5-1) + 1) // nolint: gosec
+		records[fmt.Sprintf("%s-%d.test.", baseDomain, i)] = generateTXT(rand.Intn(5-1) + 1) // nolint: gosec
```

Comment:
Same as above — try to describe why we're disabling lint.

(It might be worth just mass disabling `gosec` in `*_test.go` via `.golangci.yaml`.)

Reply:
I will do a separate PR for this as a follow-up PR.